src/Dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,190 @@
+import React, {useCallback} from 'react'
+import ReactDOM from 'react-dom'
+import styled from 'styled-components'
+import {Box, Button, ButtonDanger, ButtonPrimary, ThemeProvider, useTheme} from '..'
+import {get} from '../constants'
+import {Dialog, DialogProps, DialogHeaderProps, DialogButtonProps} from '../Dialog/Dialog'
+import {ThemeProviderProps} from '../ThemeProvider'
+
+/**
+ * Props to customize the ConfirmationDialog.
+ */
+export interface ConfirmationDialogProps {
+  /**
+   * Required. This callback is invoked when a gesture to close the dialog
+   * is performed. The first argument indicates the gesture.
+   */
+  onClose: (gesture: 'confirm' | 'cancel' | 'close-button' | 'cancel' | 'escape') => void
+
+  /**
+   * Required. The title of the ConfirmationDialog. This is usually a brief
+   * question.
+   */
+  title: string
+
+  /**
+   * The text to use for the cancel button. Default: "Cancel".
+   */
+  cancelButtonLabel?: string
+
+  /**
+   * The text to use for the confirm button. Default: "OK".
+   */
+  confirmButtonLabel?: string
+
+  /**
+   * The type of button to use for the confirm button. Default: Button.
+   */
+  confirmButtonType?: typeof Button | typeof ButtonPrimary | typeof ButtonDanger
+}
+
+const StyledConfirmationHeader = styled.header`
+  padding: ${get('space.2')};
+  display: flex;
+  flex-direction: row;
+`
+const StyledTitle = styled(Box)`
+  font-size: ${get('fontSizes.3')};
+  font-weight: 700;
+  padding: 6px ${get('space.2')};
+  flex-grow: 1;
+`
+const ConfirmationHeader: React.FC<DialogHeaderProps> = ({title, onClose, dialogLabelId}) => {
+  const onCloseClick = useCallback(() => {
+    onClose('close-button')
+  }, [onClose])
+  return (
+    <StyledConfirmationHeader>
+      <StyledTitle id={dialogLabelId}>{title}</StyledTitle>
+      <Dialog.CloseButton onClose={onCloseClick} />
+    </StyledConfirmationHeader>
+  )
+}
+const StyledConfirmationBody = styled(Box)`
+  font-size: ${get('fontSizes.1')};
+  padding: 0 ${get('space.3')} 22px ${get('space.3')};
+  color: ${get('colors.text.tertiary')};
+  flex-grow: 1;
+`
+const ConfirmationBody: React.FC<DialogProps> = ({children}) => {
+  return <StyledConfirmationBody>{children}</StyledConfirmationBody>
+}
+const StyledConfirmationFooter = styled(Box)`
+  display: flex;
+  flex-shrink: 0;
+  button {
+    font-size: ${get('fontSizes.1')};
+    flex: 1 1 0;
+    border-radius: 0;
+    border-bottom: 0;
+    margin: 0;
+    border-right: 0;
+    &:last-child {
+    }
+    &:first-child {
+      border-left: 0;
+    }
+  }
+`
+const ConfirmationFooter: React.FC<DialogProps> = ({footerButtons}) => {
```

Comment:
Do we want to create a focus zone in here?

Reply:
Yes! Accidentally left that out.